edn/src/types.rs

```diff
@@ -17,9 +17,21 @@ use symbols;
 use num::BigInt;
 use ordered_float::OrderedFloat;
 
-/// Value represents one of the allowed values in an EDN string.
+/// A wrapper type around InnerValue and Span, representing some EDN Value
+/// and the parsing offset in the original EDN string.
 #[derive(PartialEq, Eq, Hash, Clone, Debug)]
-pub enum Value {
+pub struct Value {
+    pub span: Option<Span>,
+    pub inner: InnerValue,
+}
+
+/// Span represents the current offset into the input string.
+#[derive(PartialEq, Eq, Hash, Clone, Debug)]
+pub struct Span(usize);
```

Comment:
Let's do what the Rust compiler and other places do, which is that a span is a start and an end (or a length, but I prefer end) into the input buffer.